gcc/rust/typecheck/rust-hir-type-check-implitem.h

```diff
@@ -59,12 +59,22 @@ class TypeCheckTopLevelImplItem : public TypeCheckBase
       {
 	for (auto &generic_param : function.get_generic_params ())
 	  {
-	    auto param_type
-	      = TypeResolveGenericParam::Resolve (generic_param.get ());
-	    context->insert_type (generic_param->get_mappings (), param_type);
-
-	    substitutions.push_back (
-	      TyTy::SubstitutionParamMapping (generic_param, param_type));
+	    switch (generic_param.get ()->get_kind ())
+	      {
+	      case HIR::GenericParam::GenericKind::LIFETIME:
+		// Skipping Lifetime completely until better handling.
+		break;
+		case HIR::GenericParam::GenericKind::TYPE: {
```

Comment:
Grrr, clang-format always does incorrect indent for switch case with `{ }`. I've seen it in other parts in the code. Will fix this.